engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/batch/NadelBatchHydrationByObjectId.kt

```diff
@@ -41,26 +75,46 @@ internal object NadelBatchHydrationByObjectId {
                     else -> error("Hydration actor result must be an object")
                 }
             }
-            .associateBy {
-                // We don't want to show this in the overall result, so remove it here as we use it
-                it.asMutable().remove(state.aliasHelper.getResultKey(matchStrategy.objectId))
+            .associateBy { resultNode ->
+                objectIds
+                    .map { objectId ->
+                        // We don't want to show this in the overall result, so remove it here as we use it
+                        resultNode.asMutable().remove(state.aliasHelper.getResultKey(objectId.resultId))
+                    }
```

Comment:
So this is usually where we take the batch result and generate a `Map<ObjectIdentifiedBy, BatchResult>`. We still do that here, but now the `ObjectIdentifiedBy` is a composite key (list of the specified object identified values).

e.g. given

```
identifiedBy: [
      {source: "complexIssueId.id", target: "id"}
      {source: "complexIssueId.site", target: "site"}
]
```

the key is

`listOf(resultNode["id"], resultNode["site"])`